packages/aeson/src/Conferer/Source/Aeson.hs

```diff
@@ -154,3 +178,31 @@ resultToMaybe :: Result a -> Maybe a
 resultToMaybe (Error _) = Nothing
 resultToMaybe (Success a) = Just a
 
+type RawKey = [Text]
+
+-- | Validates that a json has the correct format for keys,
+-- since Conferer 'Key's are pretty restricted.
+--
+-- The Source will work with incorrect keys, it will ignore
+-- them which is not intuitive for the user.
+validateJsonKeys :: Value -> [RawKey]
+validateJsonKeys = go []
+  where
+  go :: RawKey -> Value -> [RawKey]
+  go key value =
+    case value of
+      Object o ->
+        let
+          wrongKeys =
+            fmap (\t -> key ++ [t])
+            . filter (not . (== "_self"))
+            . filter (not . isValidKeyFragment)
+            . HashMap.keys
+            $ o
+        in wrongKeys ++ do
+        (k, v) <- HashMap.toList o
+        go (key ++ [k]) v
+      Array as -> do
+        (index :: Integer, v) <- zip [0..] $ Vector.toList as
+        go (key ++ [Text.pack $ show index]) v
```

Comment:
```suggestion
        (index :: Integer, v) <- zip [0..] $ Vector.toList as
        let subkey = key <> [Text.pack $ show index]
        go subkey v
```

The list there was confusing me a bit thinking it was something like the wrongValues 😅